Задача 1. Случайная непрерывная величина A имеет равномерное распределение на
промежутке (200, 800].
Найдите ее среднее значение и дисперсию.

In [170]:
a = 200
b = 800
M_X = (a + b) / 2
D_X = (b - a)**2 / 12
print(f'мат.ожидание равно {M_X}')
print(f'дисперсия равна {D_X}')


мат.ожидание равно 500.0
дисперсия равна 30000.0


Задача 2. О случайной непрерывной равномерно распределенной величине B известно, что ее
дисперсия равна 0.2.
Можно ли найти правую границу величины B и ее среднее значение зная, что левая
граница равна 0.5?
Если да, найдите ее.

In [171]:
a = 0.5
D_X = 0.2
# (b - 0.5)**2 / 12 = 0.2
b = 2.4**0.5 + 0.5
M_X = (a + b) / 2
print(f'мат.ожидание равно {M_X}')

мат.ожидание равно 1.2745966692414834


Задача 3. Непрерывная случайная величина X распределена нормально и задана плотностью
распределения
f(x) = (1 / (4 * sqrt(2pi))) * exp((-(x+2)**2) / 32)
Найдите:
а). M(X)
б). D(X)
в). std(X) (среднее квадратичное отклонение)

Из общей формулы плотности нормального распределения находим, что М(Х) = -2, D(X) = 16, std(X) = 4

Задача 4. Рост взрослого населения города X имеет нормальное распределение, причем, средний рост равен 174 см, а среднее квадратическое отклонение равно 8 см. посчитайте, какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
1. больше 182 см?
2. больше 190 см?
3. от 166 см до 190 см?
4. от 166 см до 182 см?
5. от 158 см до 190 см?
6. не выше 150 см или не ниже 190 см?
7. не выше 150 см или не ниже 198 см?
8. ниже 166 см?

Задачу можно решить двумя способами: без использования сторонних библиотек (numpy, scipy, pandas и пр.), а затем проверить себя с помощью встроенных функций

1. больше 182 см?

In [172]:
from scipy import stats


def z(x):
    return (x - 174) / 8


x = 182
print (z(x))

1.0


По z таблице находим вероятность и, т.к. нам надо больше 182 см вычитаем ее из 1.
P(X > 182) = 1 - 0,84134 = 0,1587

In [173]:
# проверка
1 - stats.norm.cdf(x, loc=174, scale=8)

0.15865525393145707

2. больше 190 см?

In [174]:
x = 190
print (z(x))

2.0


По z таблице находим вероятность и, т.к. нам надо больше 190 см вычитаем ее из 1.
P(X > 190) = 1 - 0,97725 = 0,0228

In [175]:
# проверка
1 - stats.norm.cdf(x, loc=174, scale=8)

0.02275013194817921

3. от 166 см до 190 см?

In [176]:
x1 = 166
x2 = 190
print (z(x1), z(x2))

-1.0 2.0


По z таблице находим вероятность для обоих X и далее рассчитываем по формуле P(a<X<b) = F(b) - F(a).
P(166<X<190) = 0,97725 - 0.15866 = 0,81859

In [177]:
# проверка
stats.norm.cdf(x2, loc=174, scale=8) - stats.norm.cdf(x1, loc=174, scale=8)

0.8185946141203637

4. от 166 см до 182 см?

In [178]:
x1 = 166
x2 = 182
print (z(x1), z(x2))

-1.0 1.0


По z таблице находим вероятность для обоих X и далее рассчитываем по формуле P(a<X<b) = F(b) - F(a).
P(166<X<182) = 0,84134 - 0.15866 = 0.68268

In [179]:
# проверка
stats.norm.cdf(x2, loc=174, scale=8) - stats.norm.cdf(x1, loc=174, scale=8)

0.6826894921370859

5. от 158 см до 190 см?

In [180]:
x1 = 158
x2 = 190
print (z(x1), z(x2))

-2.0 2.0


По z таблице находим вероятность для обоих X и далее рассчитываем по формуле P(a<X<b) = F(b) - F(a).
P(158<X<190) = 0,97725 - 0.0.02275 = 0.9545

In [181]:
# проверка
stats.norm.cdf(x2, loc=174, scale=8) - stats.norm.cdf(x1, loc=174, scale=8)

0.9544997361036416

6. не выше 150 см или не ниже 190 см?

In [182]:
x1 = 150
x2 = 190
print (z(x1), z(x2))

-3.0 2.0


По z таблице находим вероятность для обоих X и далее рассчитываем по формуле P(x<a & x>b) = 1 - F(b) + F(a).
P(x<150 & x>190) = 1 - 0,97725 + 0.0.00135 = 0.0241

In [183]:
# проверка
1 - stats.norm.cdf(x2, loc=174, scale=8) + stats.norm.cdf(x1, loc=174, scale=8)

0.0241000299798093

7. не выше 150 см или не ниже 198 см?

In [184]:
x1 = 150
x2 = 198
print (z(x1), z(x2))

-3.0 3.0


По z таблице находим вероятность для обоих X и далее рассчитываем по формуле P(x<a & x>b) = 1 - F(b) + F(a).
P(x<150 & x>198) = 1 - 0,99865 + 0.0.00135 = 0.0027

In [185]:
# проверка
1 - stats.norm.cdf(x2, loc=174, scale=8) + stats.norm.cdf(x1, loc=174, scale=8)

0.0026997960632601965

8. ниже 166 см?

In [186]:
x = 166
print (z(x))

-1.0


По z таблице находим вероятность 
P(X < 166) = 0.15866

In [187]:
# проверка
stats.norm.cdf(x, loc=174, scale=8)

0.15865525393145707

Задача 5. На сколько сигм (средних квадратичных отклонений) отклоняется рост человека,
равный 190 см, от
математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

In [188]:
M_X = 178
D_X = 25
X = 190
z = (X - M_X) / D_X**0.5
print(f'на {z} сигмы отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см')

на 2.4 сигмы отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см
